components/governance/org.wso2.carbon.governance.lcm/src/main/java/org/wso2/carbon/governance/lcm/tasks/data/JDBCLifecycleNotificationDAOImpl.java

```diff
@@ -0,0 +1,301 @@
+     * Date format used in MySQL queries.
+     */
+    private final String dateFormat = "yyyy-M-d";
+
+    /**
+     * This method is used get scheduler bran list filtering by notification method.
+     *
+     * @param registry              core registry.
+     * @return                      Array list of scheduler data objects.
+     * @throws GovernanceException  Throws when an error occurs reading data and committing transaction.
+     */
+    @Override
+    public ArrayList<LCNotification> getValidNotifications(Registry registry)
+            throws GovernanceException {
+
+        String sql = getValidNotificationQuery();
+
+        ArrayList<LCNotification> schedulerBeans = new ArrayList<LCNotification>();
+        try {
+            registry.beginTransaction();
+        } catch (RegistryException e) {
+            throw new GovernanceException("Error while reading data from registry while invoking beginTransaction for"
+                    + " query: " + sql, e);
+        }
+        JDBCDatabaseTransaction.ManagedRegistryConnection connection =
+                JDBCDatabaseTransaction.getConnection();
+        PreparedStatement preparedStatement = null;
+        try {
+            preparedStatement = connection.prepareStatement(sql);
+            preparedStatement.setString(1, getCurrentDate());
+
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                LCNotification schedulerBean = new LCNotification();
+                schedulerBean.setRegPath(resultSet.getString(LifecycleNotificationDAO.REG_PATH));
+                schedulerBean.setLcName(resultSet.getString(LifecycleNotificationDAO.REG_LC_NAME));
+                schedulerBean.setLcCheckpointId(resultSet.getString(LifecycleNotificationDAO.REG_LC_CHECKPOINT_ID));
+                schedulerBean.setNotificationDate(resultSet.getString(LifecycleNotificationDAO
+                        .REG_LC_NOTIFICATION_DATE));
+                schedulerBean.setTenantId(resultSet.getInt(LifecycleNotificationDAO.REG_TENANT_ID));
+                schedulerBeans.add(schedulerBean);
+            }
+            try {
+                registry.commitTransaction();
+            } catch (RegistryException e) {
+                throw new GovernanceException("Error while committing transaction of getting scheduler objects from "
+                        + "query: " + sql, e);
+            }
+            return schedulerBeans;
+        } catch (SQLException sqlException) {
+            try {
+                registry.rollbackTransaction();
+            } catch (RegistryException registryException) {
+                throw new GovernanceException("Error while committing transaction for query: " + sql,
+                        registryException);
+            }
+            throw new GovernanceException("SQL error while getting schedulers from query: " + sql, sqlException);
+        } finally {
+            if (preparedStatement != null) {
```

Comment:
Thanks for the comment. Close connection is handled in registry.commitTransaction() and registry.rollbackTransaction() methods using  endTransaction() inside a finally block[1][2]. So we cannot close the connection here.

Carbon4-kernel code implementation[3] of a similar implementation is listed as a refefance.

[1] https://github.com/wso2/carbon4-kernel/blob/master/core/org.wso2.carbon.registry.core/src/main/java/org/wso2/carbon/registry/core/jdbc/dataaccess/JDBCTransactionManager.java#L128
[2] https://github.com/wso2/carbon4-kernel/blob/master/core/org.wso2.carbon.registry.core/src/main/java/org/wso2/carbon/registry/core/jdbc/dataaccess/JDBCTransactionManager.java#L178
[3]https://github.com/wso2/carbon4-kernel/blob/master/core/org.wso2.carbon.registry.core/src/main/java/org/wso2/carbon/registry/core/jdbc/dao/JDBCAssociationDAO.java#L39

Thanks